extensions/shared/external-media/media-browser/index.js

```diff
@@ -99,36 +104,92 @@ function MediaBrowser( props ) {
 		);
 	};
 
+	/**
+	 * Counts how many items are in a row by checking how many of the grid's child items have a matching offsetTop.
+	 */
+	const checkColumns = () => {
+		let perRow = 1;
+
+		const items = gridEl.current.children;
+
+		if ( items.length > 0 ) {
+			const firstOffset = items[ 0 ].offsetTop;
+
+			// Check how many items have a matching offsetTop.
+			// This will give us the total number of items in a row.
+			while ( perRow < items.length && items[ perRow ].offsetTop === firstOffset ) {
+				++perRow;
+			}
+		}
+
+		setColumns( perRow );
```

Comment:
I think columns can be a mutable value as we need it only when pressing a key - no need for a state here.

Reply:
Avoid `setState` wherever possible as it will always cause a re-render. If you need the value to persist between component render lifecycles you can use a `ref` which won't cause a re-render when it's updated.

Reply:
Still very much learning React, so thanks for the tips here!